Doc/whatsnew/3.10.rst

```diff
@@ -225,6 +225,238 @@ See :class:`typing.Callable`, :class:`typing.ParamSpec`,
 
 (Contributed by Ken Jin in :issue:`41559`.)
 
+PEP 634: Structural Pattern Matching
+------------------------------------
+
+Structural pattern matching has been added in the form of a *match statement*,
+a data object (sequences, mappings, primitive data types as well as class
+instances) and *case statements* of patterns with associated actions. Pattern
+matching enables programs to extract information from complex data types,
+branch on the structure of data, and apply specific actions based on different
+forms of data.
+
+Syntax and operations
+~~~~~~~~~~~~~~~~~~~~~
+
+The generic syntax of pattern matching is::
+
+    def function(subject):
+        match {subject}:
+            case {pattern_1}:
+                {action_1}
+            case {pattern_2}:
+                {action_2}
+            case {pattern_3}:
+                {action_3}
+            case _:
+                {action_wildcard}
+
+A match statement takes an expression and compares its value to successive
+patterns given as one or more case blocks.  Specifically, pattern matching
+operates by::
+
+    1. taking a data object ``subject`` with type and shape
+    2. substituting the ``subject`` into a ``match`` statement
+    3. comparing the match subject with each pattern in a ``case`` statement
+       from top to bottom until a match is confirmed. If an exact match is not
+       confirmed, the last case, a wildcard ``_``, will be used as the matching
+       case.
+    4. executing the actions associated with the pattern of the confirmed
+       match
+
+Declarative approach
+~~~~~~~~~~~~~~~~~~~~
+
+Readers may be aware of pattern matching through the simple example of matching
+a subject (data object) to a literal (pattern) with the switch statement found
+in C, Java or JavaScript (and many other languages). Often the switch statement
+is used for comparison of an object/expression with case statements containing
+literals. 
+
+More powerful examples of pattern matching can be found in languages, such as
+Elixir. With structural pattern matching, the approach is "declarative" and
```

Comment:
```suggestion
More powerful examples of pattern matching can be found in languages such as
Scala. With structural pattern matching, the approach is "declarative" and
```
(I really hate how matching looks like in Elixir, but I was truly inspired when I saw how Scala did it. There's at least one Scala dev who will appreciate this call out.)

Reply:
I've kept Elixir and added Scala before it. There are some folks who really like the Elixir pattern matching and will be happy with the PEP addition. With you on how it looks